r2-core/src/main/java/com/linkedin/r2/message/RequestContext.java

```diff
@@ -69,7 +80,7 @@ private RequestContext(Map<String, Object> localAttrs)
    */
   public Map<String, Object> getLocalAttrs()
   {
-    return _localAttrs;
+    return Collections.unmodifiableMap(_localAttrs);
```

Comment:
If clients are making changes on the return value currently, this is a change in behavior which may break them.